### PR TITLE
[14.0][ADD] Mrp Workorder Cancel

### PR DESCRIPTION
**Pull request by Hardik-OSI**

### CODE REVIEW

File: mrp_workorder_cancel/models/mrp_workorder.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class MrpWorkorder(models.Model):
+    _inherit = "mrp.workorder"
+
+    def button_cancel(self):
+        self.ensure_one()
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "mrp_workorder_cancel.action_work_order_cancel_confirm"
+        )
+        return action
```
Comment (dreispt) [inline mrp_workorder_cancel/models/mrp_workorder.py]: FYI this is redundant - a Button can directly call a Window Action, without the need of a model method in the middle.

File: mrp_workorder_cancel/models/mrp_workorder.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    def button_workorder_cancel_undo(self):
+        work_order_ids = self.env["mrp.workorder"].browse(
+            self.env.context.get("active_ids")
+        )
+        for wo in work_order_ids:
```
Comment (dreispt) [inline mrp_workorder_cancel/models/mrp_workorder.py]: Can't you just use self?
```suggestion
        for wo in self:
```

File: mrp_workorder_cancel/views/mrp_workorder_view.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            name="inherit_id"
+            ref="mrp.mrp_production_workorder_tree_editable_view"
+        />
+        <field name="arch" type="xml">
+            <xpath expr="//tree" position="attributes">
+                <attribute name="decoration-muted">state=='cancel'</attribute>
+            </xpath>
+            <button name="button_start" position="after">
+                <button
+                    name="button_cancel"
```
Comment (dreispt) [inline mrp_workorder_cancel/views/mrp_workorder_view.xml]: This should call action_cancel and have `confirm="True"` to popup the confirmation on click.
The wizard is not really needed.

**Review by dreispt — COMMENTED** [review]
Thank you, this looks good.
I believe the goal can be achieved with significantly less code.
Can you try simplifying it?

File: mrp_workorder_cancel/views/mrp_workorder_view.xml (hunk 3 above)
Comment (Hardik-OSI) [reply mrp_workorder_cancel/views/mrp_workorder_view.xml]: Hello @dreispt 
When we add Button in List View Odoo List View dont support confirm Attribute.
So we need to add Wizard.

**Pull request closed without merging**